packages/mobile/src/navigator/NavigationService.ts

```diff
@@ -49,12 +51,59 @@ export function navigate(routeName: string, params?: NavigationParams) {
     })
 }
 
-export function navigateAfterPinEntered(routeName: string, params?: NavigationParams) {
-  navigate('Background', {
-    onUnlock() {
-      navigate(routeName, params)
-    },
-  })
+export async function ensurePincode(disableGoingBack: boolean): Promise<boolean> {
+  const pincodeType = pincodeTypeSelector(store.getState())
+
+  if (pincodeType === PincodeType.CustomPin) {
```

Comment:
What if the pincode type isn't custom?

Reply:
I've added "if" for `PincodeType.Unset`, but if it's not either of them - we pass it through (`PincodeType.PhoneAuth` is not supported and will be removed soon)